### PR TITLE
allow nice errors for auto plugin when plugin is missing or data is None

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

inventory/auto

### CODE REVIEW

**Review by jamescassell — APPROVED** [review]
shipit

File: lib/ansible/plugins/inventory/auto.py [hunk 1]
```diff
@@ -39,7 +39,10 @@ def verify_file(self, path):
     def parse(self, inventory, loader, path, cache=True):
         config_data = loader.load_from_file(path, cache=False)
 
-        plugin_name = config_data.get('plugin')
+        try:
+            plugin_name = config_data.get('plugin', None)
+        except AttributeError:
+            plugin_name = None
```
Comment (agaffney) [inline lib/ansible/plugins/inventory/auto.py:45]: Isn't this try/except redundant with the second param passed to `get()`?

Comment (mkrizek) [reply lib/ansible/plugins/inventory/auto.py:45]: Probably `config_data` might be something that does not have `get` and throw `AttributeError`?

Comment (bcoca) [reply lib/ansible/plugins/inventory/auto.py:45]: can return None, so you get an attribute error on `Nonetype does not have get`
I'm covering 2 cases with this, `config_data = None or 'plugin' not in config_data`

**Pull request merged**